rust/lance/src/dataset/schema_evolution.rs

```diff
@@ -1744,4 +1768,9 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn test_new_column_sql_to_all_nulls_transform_optimizer() {
+        // TODO: write a test to ensure the optimizer for all null sql gets used
```

Comment:
If we're OK with the general approach in this PR, I'm planning to fill in this test before we merge